File: 2019/day9.zig
const std = @import("std");
const utils = @import("utils.zig");

pub fn main() void {
    var prog = utils.loadProgram("day9_input.txt");

    // part 1
    var out = utils.Machine.init(prog).run(1) orelse unreachable;
    utils.println(out[0]);

    // part 2
    out = utils.Machine.init(prog).run(2) orelse unreachable;
    utils.println(out[0]);
}

File: 2019/day9.go
package main

import (
	"github.com/lukechampine/advent/2019/intcode"
	"github.com/lukechampine/advent/utils"
)

var input = utils.ReadInput("day9_input.txt")
var prog = utils.ExtractInts(input)

func main() {
	// part 1
	utils.Println(intcode.New(prog).Run(1)[0])
	// part 2
	utils.Println(intcode.New(prog).Run(2)[0])
}
